aemedge/templates/article/article.js

```diff
@@ -12,6 +12,29 @@ function decorate(doc) {
     main.insertBefore(tocSection, doc.querySelector('main > :nth-child(2)'));
   }
 
+  // Wrap images and captions in common container for sizing
+  const pictures = doc.querySelectorAll('p > picture');
+  pictures.forEach((pictureEl) => {
+    const parent = pictureEl.parentElement;
+    const { nextElementSibling } = parent;
+    const isCaption = nextElementSibling?.tagName.toLowerCase() === 'p'
+      && nextElementSibling.firstElementChild?.tagName.toLowerCase() === 'em';
+
+    if (isCaption) {
+      const container = domEl(
+        'figure',
+        { class: 'picture-container' },
+        pictureEl,
+        domEl(
+          'figcaption',
+          { class: 'picture-caption' },
+          nextElementSibling.firstElementChild,
+        ),
+      );
+      parent.append(container);
```

Comment:
@rjwtrmn Can we avoid wrapping picture-container around every image present on the page. This requires DOM manipulation across all images on the page.

Reply:
This should only be wrapping pictures which are immediate children of "p" tags and also followed by "p > em", but we can increase the specificity more to "main > div > p > picture"

Reply:
Hmm, I am not sure how we can achieve the caption and image being of the same width without wrapping them in the same container, since we have to infer the width from the provided image.

Reply:
Ok I understand the requirement now.